### DISCUSSION TITLE
Sequel way to add "complex" constraint

### DISCUSSION THREAD

**Discussion by felipeAH** [jeremyevans/sequel#2154] [General]
Hi,
after scratching my head a couple hours a wonder what is the sequel way to do this SQL definition:

CREATE TABLE X (
  month date CHECK ( date_part('day'. month) =  '1'))

  this:  check { Sequel.function(:date_part, 'day', Sequel.lit('month')) == '1'}
  evaluates to: check (false)

Thank you.

**Comment by adam12** [comment]
You likely want `=~` and not `==` for the comparison.

```ruby
check { Sequel.function(:date_part, 'day', Sequel.lit('month')) =~ '1' }
```
